### PR TITLE
Function dispatcher module

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

Function dispatcher to handle function assignment in cli command with multiple optional parameters.

### Added

- Dispatcher module
- Tests for the dispatcher module

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Written unit tests should pass

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/utils/dispatcher.py [hunk 1]
```diff
@@ -10,29 +8,26 @@ def __init__(self, functions: List[Callable], input_dict: Dict[str, Any]):
...
+        values_not_none: List[Any] = []
         for parameter in input_dict.keys():
             value = input_dict[parameter]
             if value is not None:
                 parameters_not_none.append(parameter)
                 values_not_none.append(value)
-        return [tuple(parameters_not_none), tuple(values_not_none)]
+        return tuple(parameters_not_none), tuple(values_not_none)
 
     def _generate_dispatch_table(self):
```
Comment (diitaz93) [inline cg/utils/dispatcher.py:26]:
```suggestion
    def _generate_dispatch_table(self) -> Dict[Tuple[str], Callable]:
```

File: cg/utils/dispatcher.py [hunk 2]
```diff
@@ -10,29 +8,26 @@ def __init__(self, functions: List[Callable], input_dict: Dict[str, Any]):
         self.dispatch_table = self._generate_dispatch_table()
```
Comment (diitaz93) [inline cg/utils/dispatcher.py:8]:
```suggestion
        self.dispatch_table: Dict[Tuple[str], Callable] = self._generate_dispatch_table()
```

**Review by diitaz93 — APPROVED** [review]
👍

**Comment by ChrOertlin** [comment]
## Description
Function dispatcher to handle function assignment in cli command with multiple optional parameters.

## Added
Dispatcher module
Tests for the dispatcher module

**Pull request merged**